plugin/keyring_vault/vault_curl.cc

```diff
@@ -0,0 +1,177 @@
+#include <algorithm>
+#include "vault_curl.h"
+#include "base64.h"
+
+namespace keyring
+{
+
+static size_t write_response_memory(void *contents, size_t size, size_t nmemb, void *userp)
```

Comment:
If that's the limit that it is Vault itself imposing, do we even have a choice? In any case, this is perfectly acceptable